doc/configuration.rst

```diff
@@ -2374,36 +2374,41 @@ etc. Similarly subclasses of 'matplotlib.axes' (e.g. 'matplotlib.axes.Axes',
 Nesting gallery sections
 ========================
 
-By default, ``nested_sections=True``.
-In this case, for each folder present in the gallery's root folder,
-Sphinx-Gallery expects to find a file ``GALLERY_HEADER.[ext]``, where
-``[ext]`` is 'txt' or an entry in ``sphinx_gallery_conf["source_suffix"]``
-(or ``README.[ext]`` for backward-compatibility). Default recommendation
-is to use ``GALLERY_HEADER.rst``. Sphinx-gallery uses that file
-to build a specific index file for this subsection.
-This index file will contain the section's description and a toctree
+``nested_sections`` lets you control how ``toctree`` sections are generated in your
+gallery ``index.rst`` files when your gallery has sub-sections (aka sub-galleries).
+This can be useful for controlling sidebar appearance.
+
+With default ``nested_sections=True``, Sphinx-Gallery will use the
+``GALLERY_HEADER.[ext]`` (or ``README.[ext]`` for backward-compatibility) files
+for the root gallery and each sub-section to build a index files for the root
+gallery and each sub-section.
+Sub-section index files will contain the sub-section's header (from the
+``GALLERY_HEADER.[ext]`` file) and a toctree
 linking to each gallery item which belongs to this subsection.
+The root gallery's main ``index.rst`` file will contain, in sequence:
 
-Eventually, the gallery's main index files will contain the gallery's
-description and a toctree linking to each subsections's index file.
+* root gallery header followed by gallery thumbnails,
+* a toctree linking to each gallery item in the root gallery,
+* sub-section header followed by sub-section thumbnails, for all subsections,
+* a final toctree linking to each sub-section's index file.
 
-With this behaviour, generated file structure and toctrees mimic that of
-the original gallery folder. This is useful to generate sidebars with
+With this behaviour the generated file structure and toctrees mimic that of
+the root gallery folder. This is useful for generating sidebars with
 nested sections representing the gallery's file structure.
 
 .. note::
 
-    When ``nested_sections=True``, gallery items located in the gallery's root folder
-    should be move to a new subfolder, otherwise the sidebar
+    When ``nested_sections=True``, gallery examples located in the gallery's root
+    folder should be moved to a new subfolder, otherwise the sidebar
     might not behave as expected (due to the fuzzy toctree structure).
```

Comment:
@alexisthual what do you mean by 'fuzzy' toctree structure?

Also do you know exactly what problem not moving examples to their own sub-folder causes? 

Indeed Sphinx-Gallery has root gallery example files and this seems to play well with PST theme,

Reply:
I think I may have a better idea of the problem, it seems that having 2 toctrees in the `index.rst` may cause duplication (??) in the sidebar? Regardless, I think that putting all examples in their own sub-folder and none in the root gallery folder results in a single toctree in the gallery root `index.rst`.

Reply:
Hopefully the change I made is correct, let me know @alexisthual !

Reply:
I am sorry, I don't remember what I exactly meant by 'fuzzy toctree structure', but I think your second comment here gives the right intuition.